Lib/test/test_gettext.py

```diff
@@ -730,25 +730,28 @@ def test_gnu_translations_fallback(self):
                          'npgettext: context, foo, foos, 1')
 
     def test_nested_fallbacks(self):
-        class NestedFallback(gettext.NullTranslations):
+        class NestedFallback1(gettext.NullTranslations):
             def gettext(self, message):
-                return 'fallback 1'
+                if message == 'foo':
+                    return 'fallback 1'
+                return super().gettext(message)
 
         class NestedFallback2(gettext.NullTranslations):
-            def pgettext(self, context, message):
+            def gettext(self, message):
                 return 'fallback 2'
 
-        class NestedFallback3(gettext.NullTranslations):
-            def ngettext(self, msgid1, msgid2, n):
-                return 'fallback 3'
-
+        fallback1 = NestedFallback1()
+        fallback2 = NestedFallback2()
         t = gettext.NullTranslations()
-        t.add_fallback(NestedFallback())
-        t.add_fallback(NestedFallback2())
-        t.add_fallback(NestedFallback3())
+
+        t.add_fallback(fallback1)
+        self.assertEqual(t._fallback, fallback1)
+        t.add_fallback(fallback2)
+        self.assertEqual(t._fallback, fallback1)
+        self.assertEqual(t._fallback._fallback, fallback2)
```

Comment:
It is better to test not private attributes, but behavior. Checking `fallback1.gettext('bar')` should be enough.

Reply:
Changed in [45b152f](https://github.com/python/cpython/pull/130321/commits/45b152f06c49149d695dcbb3287c5c39c35ba88f) :)